indico/modules/events/abstracts/forms.py

```diff
@@ -146,19 +146,46 @@ class AbstractJudgmentForm(IndicoForm):
     session = QuerySelectField(_("Session"), [HiddenUnless('judgment', AbstractAction.accept)],
                                get_label='title', allow_blank=True, blank_text=_("Choose a session..."),
                                description=_("The generated contribution will be allocated in this session"))
+    duplicate_of = AbstractField(_("Duplicate of"),
+                                 [HiddenUnless('judgment', AbstractAction.mark_as_duplicate), DataRequired()],
+                                 description=_("The current abstract will be marked as duplicate of the selected one"))
     merged_into = AbstractField(_("Merge into"), [HiddenUnless('judgment', AbstractAction.merge), DataRequired()],
                                 description=_("The current abstract will be merged onto the selected one"))
-    merge_persons = BooleanField(_("Merge persons"), [HiddenUnless('judgment', AbstractAction.merge), DataRequired()],
+    merge_persons = BooleanField(_("Merge persons"), [HiddenUnless('judgment', AbstractAction.merge)],
                                  description=_("Authors and speakers of the current abstract will be added to the "
                                                "selected one"))
-    duplicate_of = AbstractField(_("Duplicate of"),
-                                 [HiddenUnless('judgment', AbstractAction.mark_as_duplicate), DataRequired()],
-                                 description=_("The current abstract will be marked as duplicate of the selected one"))
     judgment_comment = TextAreaField(_("Comment"), render_kw={'placeholder': _("Leave a comment for the submitter...")})
     # TODO: show only if notifications apply?
     send_notifications = BooleanField(_("Send notifications to submitter"), default=True)
 
     def __init__(self, *args, **kwargs):
+        super(AbstractJudgmentFormBase, self).__init__(*args, **kwargs)
+        self.session.query = Session.query.with_parent(self.event).order_by(Session.title)
+        self.accepted_track.query = Track.query.with_parent(self.event).order_by(Track.title)
+        self.accepted_contrib_type.query = (ContributionType.query
+                                            .with_parent(self.event)
+                                            .order_by(ContributionType.name))
+
+    @property
+    def split_data(self):
+        abstract_data = self.data
+        judgment = abstract_data.pop('judgment')
+        special_data = {
+            'judgment': judgment if judgment else AbstractAction[request.form['judgment']],
```

Comment:
In case we judge the abstract from the abstract management page, the judgment is in abstract_data. If we judge it from the abstract list, it's in the request.form